src/app/page/UserProfile/LinkPhoneAccount.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Grid, Typography, TextField, Input } from "@material-ui/core";
+import { Button } from "../../component";
+import EditIcon from "@material-ui/icons/Edit";
+import { Card } from "../../layout";
+
+function LinkPhoneAccount({ auth, data, errorHandler }) {
+  const [phoneNumber, updatePhoneNumber] = useState("");
+
+  async function onPhoneClick() {
+    auth.useDeviceLanguage();
+    const verifier = new auth.RecaptchaVerifier("phone-number-link", { size: "invisible" });
+    try {
+      const confirmationResult = await auth.currentUser.linkWithPhoneNumber(phoneNumber, verifier);
+      const verificationCode = window.prompt(
+        "Please enter the verification code that was sent to your mobile device."
+      );
+      await confirmationResult.confirm(verificationCode);
+    } catch (error) {
+      errorHandler(error);
+    }
+  }
+
+  return data ? (
+    <>
+      <Input
+        label="your phone number..."
+        inputProps={{ "aria-label": "phoneNumber" }}
+        defaultValue={data.phoneNumber}
+        disabled
+      />
+
+      <Button
+        id="phone-number-link"
+        onClick={onPhoneClick}
+        color="default"
+        aria-label="change phone number"
+      >
+        <EditIcon /> Change Phone Number
+      </Button>
+
+      <Button id="phone-number-link" onClick={onPhoneClick} aria-label="change phone number">
+        Disconnect
+      </Button>
+    </>
+  ) : (
+    <Grid container direction="column" spacing={2}>
```

Comment:
We can reuse SMSLogin.jsx here.

Reply:
Yeah, not possible, there is going to be unlink and just other stuff as well